curated_transformers/tokenization/gpt_neox_tokenizer.py

```diff
@@ -1,49 +1,35 @@
-from typing import Any, Type, TypeVar
-from curated_tokenizers import ByteBPEProcessor
+from typing import Any, Tuple, Type, TypeVar, cast
 import json
-from pathlib import Path
 
 from .bbpe_tokenizer import ByteBPETokenizer
-from .hf_hub import FromPretrainedHFTokenizer
+from .hf_hub import FromHFHub, FromPretrainedHFTokenizer
 
 # Only provided as typing.Self in Python 3.11+.
 Self = TypeVar("Self", bound="GPTNeoXTokenizer")
 
 
-class GPTNeoXTokenizer(ByteBPETokenizer, FromPretrainedHFTokenizer):
+class GPTNeoXTokenizer(ByteBPETokenizer, FromHFHub, FromPretrainedHFTokenizer):
     """GPT-NeoX tokenizer (Black et al., 2022).
 
     The GPT-NeoX tokenizer uses byte-level byte pair encoding.
     """
 
     @classmethod
-    def from_files(
-        cls: Type[Self],
-        *,
-        vocab_path: Path,
-        merges_path: Path,
-    ) -> Self:
-        """Construct a tokenizer from the vocabulary and merges files.
-
-        vocab_path (Path): path to the vocabulary file.
-        merges_path (Path): path to the merges file.
-        """
-        processor = ByteBPEProcessor.load_from_files(
-            vocab=vocab_path, merges=merges_path
-        )
-        return cls(processor=processor)
+    def _convert_hf_tokenizer_json(cls: Type[Self], *, hf_tokenizer: Any) -> Self:
+        # TODO: Seems like we can't easily probe the model type?
```

Comment:
I [see](https://huggingface.co/EleutherAI/gpt-neox-20b/raw/main/tokenizer.json) a `type` entry in `hf_tokenizer["model"]`?

Reply:
But the type is `BPE`, which is not really specific. But I guess it's better than nothing, so I'll add that.

Reply:
Though thinking more about this, the NeoX tokenizer is pretty much a plain Byte BPE tokenizer, so I'll move the loading methods there.